packages/netlify-cms-widget-list/src/ListControl.js

```diff
@@ -200,10 +200,13 @@ export default class ListControl extends React.Component {
     return (fieldName, newValue, newMetadata) => {
       const { value, metadata, onChange, field } = this.props;
       const collectionName = field.get('name');
-      const newObjectValue =
-        this.getValueType() !== valueTypes.SINGLE
-          ? this.getObjectValue(index).set(fieldName, newValue)
-          : newValue;
+      const listFieldObjectWidget = field.getIn(['field', 'widget']) === 'object';
```

Comment:
I would expect `field.widget` to almost always be `"list"` - what am I missing?

Reply:
The fix here is to save the value just like we do for `list`  multiple `fields`  i.e using widget name as key  `{widget_name: value}`.
So `this.getValueType() === valueTypes.SINGLE && listFieldObjectWidget` should match config setup as described in the issue.